lib/cron/synth-switch.ts

```diff
@@ -41,14 +79,178 @@ const handler: ScheduledHandler = async (_event: EventBridgeEvent<string, void>)
+    const trade: SynthSwitchRequestBody = {
+      inputToken: order.tokenin,
+      inputTokenChainId: parseInt(order.tokeninchainid),
+      outputToken: order.tokenout,
+      outputTokenChainId: parseInt(order.tokenoutchainid),
+      type: String(tradeType),
+      amount: tradeType == TradeType.EXACT_INPUT ? order.classic_amountin : order.classic_amountout,
+    };
+    const key = SwitchRepository.getKey(trade);
+    let hasPriceImprovement: boolean;
+    if (tradeType == TradeType.EXACT_INPUT) {
+      hasPriceImprovement = BigNumber.from(order.settledAmountOut).gt(order.classic_amountoutgasadjusted);
+    } else {
+      hasPriceImprovement = BigNumber.from(order.classic_amountingasadjusted).gt(order.settledAmountIn);
+    }
+    // can add more conditionals here
+    const result = hasPriceImprovement;
+    return { key, result };
+  }
+
+  async function updateSynthSwitchRepository(configs: TokenConfig[], result: ResultRowType[]) {
+    // match configs to results
+    const configMap: {
+      [key: string]: ResultRowType[];
+    } = {};
+    result.map((row) => {
+      const key = `${row.tokenin}#${row.tokeninchainid}#${row.tokenout}#${row.tokenoutchainid}`;
+      if (key in configMap) {
+        configMap[key].push(row);
+      } else {
+        configMap[key] = [row];
+      }
+    });
+
+    for (const config of configs) {
+      // totalTrades is both ExactIn and ExactOut
+      const ordersForConfig =
+        configMap[
+          `${config.inputToken}#${config.inputTokenChainId}#${config.outputToken}#${config.outputTokenChainId}`
+        ];
+      // build trade objects differentiating between ExactIn and ExactOut
+      let tradeOutcomesByKey: {
+        [key: string]: {
+          pos: number;
+          neg: number;
+        };
+      } = {};
+      for (const order of ordersForConfig) {
+        const { key, result } = hasPositiveTradeOutcome(order);
+        if (!(key in tradeOutcomesByKey)) {
+          tradeOutcomesByKey[key] = {
+            pos: 0,
+            neg: 0,
+          };
+        }
+        if (result) {
+          tradeOutcomesByKey[key].pos++;
+        } else {
+          tradeOutcomesByKey[key].neg++;
+        }
+      }
 
+      Object.keys(tradeOutcomesByKey).forEach(async (key) => {
+        const { pos, neg } = tradeOutcomesByKey[key];
+        const totalOrders = pos + neg;
+        if (totalOrders >= MINIMUM_ORDERS) {
+          if (neg / totalOrders >= DISABLE_THRESHOLD) {
+            log.info(
+              {
+                key,
+                ordersWithNegativeOutcome: neg,
+                ordersWithPositiveOutcome: pos,
+                totalOrders,
+              },
+              'Disabling synthethics for trade'
+            );
+            // TODO: update tradeSizes with new TokenConfig schema
+            await synthSwitchEntity.putSynthSwitch(SwitchRepository.parseKey(key), config.tradeSizes[0], false);
```

Comment:
trying to think through a potential edge case: say we originally have { in: USDC, inChain: 1, out: UNI, outChain: 1, type: ExactIn, lower: 100,000 } in the config and enabled = false. If somehow we lower `lower` to 10,000 and the cron decided to put `enabled = true` we'd then have two rows with the same partition key. Currently we query the table in reverse order meaning the row with `enabled = false`  will be returned.

maybe it makes more sense to remove the sort key (lower) from the table? so only one row (or none) can exist

Reply:
> so only one row (or none) can exist
i was under the impression that this was true in the current implementation but yeah I see what you mean. I thought that the default behavior if the partition key exists is that it would update it with the new sort key value?